omplbindings/bindings.go

```diff
@@ -96,95 +86,84 @@ func ComputePositions(pos []float64) *C.struct_pose {
-	if err != nil {
-		fmt.Println(err)
-	}
-
-	ik_sol_len := len(solutions[0].Q())
+	// solutions, err := motionplan.BestIKSolutions(
+	// 	context.Background(),
+	// 	logger,
+	// 	sceneFS,
+	// 	scene.RobotFrame,
+	// 	scene.Start,
+	// 	cToPose(targetPoseC),
+	// 	scene.WorldState,
+	// 	1,
+	// 	options,
+	// )
+	// if err != nil {
+	// 	fmt.Println(err)
+	// }
+	solution := make([]referenceframe.Input, 6)
+	ik_sol_len := len(solution)
 	ik_sol_ptr := C.malloc(C.size_t(ik_sol_len) * C.size_t(unsafe.Sizeof(C.double(0))))
 	ik_solution := (*[1 << 30]C.double)(ik_sol_ptr)[:ik_sol_len:ik_sol_len]
 	for i := 0; i < ik_sol_len; i++ {
-		ik_solution[i] = C.double(solutions[0].Q()[i].Value)
+		ik_solution[i] = C.double(solution[i].Value)
 	}
 	return (*C.double)(ik_sol_ptr)
 }
 
 //export ValidState
 func ValidState(pos []float64) bool {
-	cInput := &motionplan.ConstraintInput{StartInput: referenceframe.FloatsToInputs(pos), Frame: sceneFS.Frame(testArmFrame)}
-	for _, cons := range constraints {
-		pass, _ := cons(cInput)
-		if !pass {
-			return false
-		}
-	}
+	// cInput := &motionplan.ConstraintInput{StartInput: referenceframe.FloatsToInputs(pos), Frame: sceneFS.Frame(testArmFrame)}
+	// for _, cons := range constraints {
+	// 	pass, _ := cons(cInput)
+	// 	if !pass {
+	// 		return false
+	// 	}
+	// }
 	return true
 }
 
 //export VisualizeOMPL
-func VisualizeOMPL(inputs [][]float64) {
-	plan := make([][]referenceframe.Input, 0)
-	nSteps := 0
-	for i, input := range inputs {
-		pStep := referenceframe.FloatsToInputs(input)
-		plan = append(plan, pStep)
-		if i < len(inputs) - 1 {
-			nextStep := referenceframe.FloatsToInputs(inputs[i+1])
-			for j := 1; j <= nSteps; j++ {
-				step := referenceframe.InterpolateInputs(pStep, nextStep, float64(j)/float64(nSteps))
-				plan = append(plan, step)
-			}
-		}
-	}
-	visualization.VisualizePlan(scene.RobotFrame, plan, scene.WorldState)
+func VisualizeOMPL(inputs [][]float64) error {
+	// plan := make([][]referenceframe.Input, 0)
+	// nSteps := 0
+	// for i, input := range inputs {
+	// 	pStep := referenceframe.FloatsToInputs(input)
+	// 	plan = append(plan, pStep)
+	// 	if i < len(inputs)-1 {
+	// 		nextStep := referenceframe.FloatsToInputs(inputs[i+1])
+	// 		for j := 1; j <= nSteps; j++ {
+	// 			step := referenceframe.InterpolateInputs(pStep, nextStep, float64(j)/float64(nSteps))
+	// 			plan = append(plan, step)
+	// 		}
+	// 	}
+	// }
+	// worldState, err := referenceframe.WorldStateToProtobuf(scene.WorldState)
+	// if err != nil {
+	// 	return nil
+	// }
+	// return visualization.VisualizePlan(scene.RobotFrame, plan, worldState)
+	return nil
 }
 
 //export Init
-func Init(name string) {
+func Init(name string) error {
 	sceneFS = referenceframe.NewEmptySimpleFrameSystem("")
 
 	// setup scenes in global vars
 	initFunc, ok := allScenes[name]
 	if !ok {
-		fmt.Println("Scene '", name, "'does not exist!")
-		return
+		msg := "scene '" + name + "'does not exist"
+		fmt.Println(msg)
```

Comment:
[nit] Could we start using the logger here instead?

Reply:
changed